### PR TITLE
Add scripts to store git history for releases, locks, deployments

### PR DESCRIPTION
**Pull request by tameremad**

### CODE REVIEW

File: infrastructure/scripts/metadata-migration/create-metadata-app-version.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -eu # protection
```
Comment (hannesg) [inline infrastructure/scripts/metadata-migration/create-metadata-app-version.sh:3]: Please use this line:

```
set -eou pipefail
```

File: infrastructure/scripts/metadata-migration/create-metadata-locks.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  echo -----------------Application Locks------------------------
+
+  echo -----------------Environment Locks------------------------
+  # If $env has locks
+  if [ -d "$env"/locks ]; then
+    echo Adding metadata to "$(ls "$env"/locks | wc -l)" locks in "$env"
+    find "$env"/locks  -maxdepth 1 -mindepth 1 -type f -print0 | while IFS= read -r -d '' lock
+    do
+      echo Lock ID: "$(basename "$lock")" - Lock Message: "$(cat "$lock")"
+      LOCK_DIR="$(dirname "$lock")"/lock_"$(basename "$lock")"
```
Comment (hannesg) [inline infrastructure/scripts/metadata-migration/create-metadata-locks.sh:39]: Why not just turn the file into a directory? The prefix `lock_` shouldn't be necessary.

File: infrastructure/scripts/metadata-migration/create-metadata-app-version.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+echo --------------------Versions------------------------------
+for env in environments/* # $(find environments  -maxdepth 1 -mindepth 1 -type d)
+do
+  for app in "$env"/applications/*
+  do
+    # If $app has version
+    echo Application Version: "$app"
+    if [ -d "$app"/version ]; then
+      git log -1 --date=iso-strict --format="%ad" -- "$app"/version > "$app"/deploy_date
```
Comment (hannesg) [inline infrastructure/scripts/metadata-migration/create-metadata-app-version.sh:14]: I recall that we discussed removing this date entirely. Do we still need it?

File: infrastructure/scripts/metadata-migration/create-metadata-app-version.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (hannesg) [inline infrastructure/scripts/metadata-migration/create-metadata-app-version.sh:1]: Please use this:

```
#!/usr/bin/env bash
```

**Review by hannesg — APPROVED** [review]

File: infrastructure/scripts/metadata-migration/create-metadata-app-version.sh (hunk 3 above)
Comment (tameremad) [reply infrastructure/scripts/metadata-migration/create-metadata-app-version.sh:14]: I thought we could use it in the future as the deploy date. and as this is a one-time migration, creating this now would be better? 

also, the backend can write the deploy date for newer deployments before we find a good way to show it in the UI. 

I think the feature is nice but it's not urgent. 
in the future:
* we can uncomment this field in the API and read the date from the repository. 
* then it's available in the frontend. 
* then we find a good way to show this in the UI.

**Pull request merged**